src/vm/hints/secp/ec_utils.rs

```diff
@@ -255,6 +258,151 @@ pub fn ec_double_assign_new_y(vm: &mut VirtualMachine) -> Result<(), VirtualMach
     Ok(())
 }
 
+/*
+Implements hint:
+%{
+    from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+
+    slope = pack(ids.slope, PRIME)
+    x0 = pack(ids.point0.x, PRIME)
+    x1 = pack(ids.point1.x, PRIME)
+    y0 = pack(ids.point0.y, PRIME)
+
+    value = new_x = (pow(slope, 2, SECP_P) - x0 - x1) % SECP_P
+%}
+*/
+pub fn fast_ec_add_assign_new_x(
+    vm: &mut VirtualMachine,
+    ids: &HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    //ids.slope
+    let slope_reloc = get_relocatable_from_var_name(
+        "slope",
+        ids,
+        &vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )?;
+
+    let (slope_d0, slope_d1, slope_d2) = (
+        vm.memory.get_integer(&slope_reloc)?,
+        vm.memory.get_integer(&(&slope_reloc + 1))?,
+        vm.memory.get_integer(&(&slope_reloc + 2))?,
+    );
+
+    //ids.point0
+    let point0_reloc = get_relocatable_from_var_name(
+        "point0",
+        ids,
+        &vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )?;
+
+    let (point0_x_d0, point0_x_d1, point0_x_d2, point0_y_d0, point0_y_d1, point0_y_d2) = (
+        vm.memory.get_integer(&point0_reloc)?,
+        vm.memory.get_integer(&(&point0_reloc + 1))?,
+        vm.memory.get_integer(&(&point0_reloc + 2))?,
+        vm.memory.get_integer(&(&point0_reloc + 3))?,
+        vm.memory.get_integer(&(&point0_reloc + 4))?,
+        vm.memory.get_integer(&(&point0_reloc + 5))?,
+    );
+
+    //ids.point1.x
+    let point1_reloc = get_relocatable_from_var_name(
+        "point1",
+        ids,
+        &vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )?;
+
+    let (point1_x_d0, point1_x_d1, point1_x_d2) = (
+        vm.memory.get_integer(&point1_reloc)?,
+        vm.memory.get_integer(&(&point1_reloc + 1))?,
+        vm.memory.get_integer(&(&point1_reloc + 2))?,
+    );
+
+    let slope = pack(slope_d0, slope_d1, slope_d2, &vm.prime);
+    let x0 = pack(point0_x_d0, point0_x_d1, point0_x_d2, &vm.prime);
+    let x1 = pack(point1_x_d0, point1_x_d1, point1_x_d2, &vm.prime);
+    let y0 = pack(point0_y_d0, point0_y_d1, point0_y_d2, &vm.prime);
+
+    let value = (slope.pow(2).mod_floor(&SECP_P) - &x0 - x1).mod_floor(&SECP_P);
```

Comment:
Due to the equivalence in congruence of substraction, you could save one `mod_floor` by only performing it at the end:
```suggestion
    let value = (slope.pow(2) - &x0 - x1).mod_floor(&SECP_P);
```

Reply:
Nice, I missed this.
Done